Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py

```diff
@@ -112,12 +125,12 @@ def delete_policy(self, policy_name: str, resource_group_name: str):
 
 
 def test_connection(client: AzureWAFClient, params: Dict):
-    if params.get('self_deployed', False) and not params.get('auth_code'):
-        return_error('You must enter an authorization code in a self-deployed configuration.')
-    client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
-    # should return error if user does not have permissions for the service.
-    client.get_policy_list_by_resource_group_name(client.resource_group_name)
-    return '✅ Great Success!'
+    if demisto.params().get('auth_type') == 'Device':
+        client.get_policy_list_by_resource_group_name(client.resource_group_name)  # If fails, MicrosoftApiModule returns an error
+        return CommandResults(readable_output='✅ Success!')
+    else:
+        client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
+        return CommandResults(readable_output='✅ Success!')
```

Comment:
```suggestion
    if demisto.params().get('auth_type') == 'Device':
        client.get_policy_list_by_resource_group_name(client.resource_group_name)  # If fails, MicrosoftApiModule returns an error
    else:
        client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
return CommandResults(readable_output='✅ Success!')
```

Reply:
not sure if this came out normal or not. I just moved `return CommandResults(readable_output='✅ Success!')` out of the if else block